vyper/builtin_functions/functions.py

```diff
@@ -1866,13 +1971,21 @@ class Empty(_BuiltinFunction):
     _inputs = [("typename", "*")]
 
     def fetch_call_return(self, node):
-        validate_call_args(node, 1)
-        type_ = get_type_from_annotation(node.args[0], DataLocation.MEMORY)
+        type_ = self.infer_arg_types(node)[0].typedef
         return type_
 
+    def infer_arg_types(self, node):
+        validate_call_args(node, 1)
+        input_typedef = TypeTypeDefinition(
+            get_type_from_annotation(node.args[0], DataLocation.MEMORY)
+        )
+        # Annotate argument AST node because empty is skipped in visit_Call
```

Comment:
maybe we should not skip `empty` in `visit_Call`

Reply:
If we annotate `empty` in `visit_Call`, `visit_Subscript` will throw an error for arrays and structs as these are type definitions but it currently expects a value (not a type) when it makes a call to `get_exact_type_from_node` to get the `base_type`.

Reply:
`empty` should work, and `range` should be removed from `visit_Call` branch ideally

Reply:
so i think we can actually get rid of this override entirely, right?

Reply:
we still need the override in order to annotate the `TypeTypeDefinition`.